bindings/bind_py_structure_manager.cc

```diff
@@ -25,489 +25,646 @@
  * Boston, MA 02111-1307, USA.
  */
 
-#include "bind_include.hh"
-
-using namespace rascal;  // NOLINT
-
-/**
- * In this file SMI stands for StructureManagerImplementation.
- */
-
-// Collection of typedefs to simplify the bindings
-template <typename SMI>
-using LayerByOrder = typename SMI::traits::LayerByOrder;
-
-template <typename SMI, size_t Order>
-struct HelperLayer {
-  static constexpr size_t layer{
-      compute_cluster_layer<Order>(LayerByOrder<SMI>{})};
-};
-
-template <typename SMI, size_t Order>
-using ClusterRefkey_t = ClusterRefKey<Order, HelperLayer<SMI, Order>::layer>;
-
-template <typename SMI, size_t Order>
-using ClusterRef_t = typename StructureManager<SMI>::template ClusterRef<Order>;
-
-template <typename SMI, size_t Order>
-using PyClusterRef =
-    py::class_<ClusterRef_t<SMI, Order>, ClusterRefkey_t<SMI, Order>>;
-
-template <typename StructureManagerImplementation>
-using PyManager = py::class_<StructureManagerImplementation,
-                             StructureManager<StructureManagerImplementation>,
-                             std::shared_ptr<StructureManagerImplementation>>;
-
-//! Bind a ClusterRef (to enable the acces to properties such as distances)
-template <size_t Order, size_t Layer>
-void add_cluster_ref(py::module & m) {
-  std::string cluster_parent_name =
-      internal::GetBindingTypeName<ClusterRefKey<Order, Layer>>();
-
-  py::class_<ClusterRefKey<Order, Layer>, ClusterRefBase>(
-      m, cluster_parent_name.c_str());
-}
-
-//! Bind many cluster refs of a given order using template recursion
-template <size_t Order, size_t Layer, size_t LayerEnd>
-struct add_cluster_refs {
-  //! starts recursion
-  static void static_for(py::module & m) {
-    add_cluster_ref<Order, Layer>(m);
-    add_cluster_refs<Order, Layer + 1, LayerEnd>::static_for(m);
-  }
-};
-
-//! Stop the recursion
-template <size_t Order, size_t LayerEnd>
-struct add_cluster_refs<Order, LayerEnd, LayerEnd> {
-  static void static_for(py::module &) {}
-};
-
-//! Bind clusters of different orders
-//! (up to quadruplet)
-template <size_t Order, typename SMI>
-decltype(auto) add_cluster(py::module & m) {
-  using ClusterRef = ClusterRef_t<SMI, Order>;
-  using ClusterRefKey = ClusterRefkey_t<SMI, Order>;
-
-  std::string cluster_name = internal::GetBindingTypeName<SMI>();
-
-  static_assert(Order < 5, "Can't bind more than quadruplet as it is.");
-
-  if (Order == 1) {
-    cluster_name += std::string(".Center");
-  } else if (Order == 2) {
-    cluster_name += std::string(".Pair");
-  } else if (Order == 3) {
-    cluster_name += std::string(".Triplet");
-  } else if (Order == 4) {
-    cluster_name += std::string(".Quadruplet");
+#include "bind_py_structure_manager.hh"
+#include "bind_py_representation_calculator.hh"
+
+namespace rascal {
+  /**
+   * In this file SMI stands for StructureManagerImplementation.
```

Comment:
Is this comment meant to be parsed by Doxygen? I'm guessing not, in which case it shouldn't be in Doxygen format.